pkg/statsd/parser.go

```diff
@@ -85,8 +85,8 @@ func (dp *DatagramParser) RunMetricsContext(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-flushed:
-			statser.Gauge("parser.metrics_received", float64(atomic.LoadUint64(&dp.metricsReceived)), nil)
-			statser.Gauge("parser.events_received", float64(atomic.LoadUint64(&dp.eventsReceived)), nil)
+			statser.Report("parser.metrics_received", float64(atomic.LoadUint64(&dp.metricsReceived)), nil)
```

Comment:
These will still accumulate.  I think a better approach is to pass a pointer, and then have `Report` do the `LoadUint64` vs `SwapUint64(0)`.

Nothing actually needs it to be a float64, and if we define it as a "counter type thing", nothing needs to.

Reply:
- [ ] Pass pointer, and have atomic ops in the statser

Reply:
So keep the same logic that Report can either be a Gauge or Count but have it accept a pointer to a value that it can reset on each report?